src/backend/distributed/planner/multi_router_planner.c

```diff
@@ -871,6 +904,137 @@ ErrorIfInsertPartitionColumnDoesNotMatchSelect(Query *query, RangeTblEntry *inse
 
 
 /*
+ * FindOriginalTableId recursively traverses query tree to find actual relation
+ * id that columnExpression refers to. It returns InvolidOid if Var is a non-relational
+ * or computed value from an inner subquery or cte. This is very similar to
+ * IsPartitionColumnRecursive.
+ */
+Oid
+FindOriginalRelationId(Expr *columnExpression, List *parentQueryList, Query *query,
+					   bool *isPartitionColumn)
+{
+	Var *candidateColumn = NULL;
+	List *rangetableList = query->rtable;
+	Index rangeTableEntryIndex = 0;
+	RangeTblEntry *rangeTableEntry = NULL;
+	Expr *strippedColumnExpression = (Expr *) strip_implicit_coercions(
+		(Node *) columnExpression);
+	Oid relationId = InvalidOid;
+	*isPartitionColumn = false;
+
+	if (IsA(strippedColumnExpression, Var))
+	{
+		candidateColumn = (Var *) strippedColumnExpression;
+	}
+	else if (IsA(strippedColumnExpression, FieldSelect))
+	{
+		FieldSelect *compositeField = (FieldSelect *) strippedColumnExpression;
+		Expr *fieldExpression = compositeField->arg;
+
+		if (IsA(fieldExpression, Var))
+		{
+			candidateColumn = (Var *) fieldExpression;
+		}
+		else
+		{
+			ereport(ERROR, (errcode(ERRCODE_FEATURE_NOT_SUPPORTED),
+							errmsg("cannot push down this subquery"),
+							errdetail("Only references to column fields are supported")));
```

Comment:
This was from the original IsPartitionColumnRecursive function. However, none of the regression tests catch this part. Removing.